src/SFA.DAS.ApplyService.Web/Extensions/TempDataDictionaryExtensions.cs

```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Newtonsoft.Json;
+
+namespace SFA.DAS.ApplyService.Web.Extensions
+{
+    public static class TempDataDictionaryExtensions
+    {
+        public static T Get<T>(this ITempDataDictionary tempData) where T : class
```

Comment:
I don't think we use structs, but isn't this something we should have? You could then check for ISerializable too

Reply:
Sorry, I don't follow - what change are you asking for?

Reply:
T is constrained to class. It is possible to have structs that implement ISerializable

Hence I'd expect T to be new() and do a check that it is a class type or implements ISerializable

If it's overkill don't worry about it

Reply:
actually ignore me. If we make this too flexible the code checks will become hideous and we're allowing other devs to have types that aren't really allowed in viewmodels